File: src/main/java/com/vaadin/demo/component/upload/UploadAutoUploadDisabled.java
package com.vaadin.demo.component.upload;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.upload.Upload;
import com.vaadin.flow.component.upload.receivers.MultiFileMemoryBuffer;
import com.vaadin.flow.router.Route;

@Route("upload-auto-upload-disabled")
public class UploadAutoUploadDisabled extends Div {

    public UploadAutoUploadDisabled() {
        MultiFileMemoryBuffer buffer = new MultiFileMemoryBuffer();
        // tag::snippet[]
        Upload upload = new Upload(buffer);
        upload.setAutoUpload(false);

        UploadExamplesI18N i18n = new UploadExamplesI18N();
        i18n.getAddFiles()
                .setMany("Select Files...");
        upload.setI18n(i18n);
        // end::snippet[]

        upload.getElement().executeJs("this.files = this.createFakeFilesUploadAutoUploadDisabled()"); // hidden-source-line

        add(upload);
    }

    public static class Exporter extends // hidden-source-line
            DemoExporter<UploadAutoUploadDisabled> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/upload/upload-auto-upload-disabled.ts
import 'Frontend/demo/init'; // hidden-source-line
import './upload-demo-helpers'; // hidden-source-line
import '@vaadin/upload';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/generated/theme';
import { createFakeFilesUploadAutoUploadDisabled } from './upload-demo-mock-files'; // hidden-source-line

@customElement('upload-auto-upload-disabled')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  // tag::snippet[]
  private uploadI18n = {
    addFiles: {
      many: 'Select Files...',
    },
  };

  protected override render() {
    return html`
      <vaadin-upload
        no-auto
        .i18n="${this.uploadI18n}"
        .files="${createFakeFilesUploadAutoUploadDisabled() /* hidden-source-line */}"
      ></vaadin-upload>
    `;
  }
  // end::snippet[]
}
